### DISCUSSION TITLE
Import CSV or VTK file of pore body and throat in openPNM

### DISCUSSION THREAD

**Discussion by Soheil1993** [PMEAL/OpenPNM#2417] [Q&A]
Dear Dr. @jgostick 

I extracted the pore body data in both CSV (x,y,z,r) and VTK formats and pore throat data in both CSV (pore id1,pore id2,r) and VTK formats from Yade. Now I want to use openPNM for obtaining the permeability of the media.

Would you please let me know how I should modify the files and import them into the openPNM?

Should I combine them or just import the pore throat file for permeability?

Best regards,
Soheil

**Comment by ma-sadeghi** [comment]
You can use `openpnm.io.from_csv` but that requires you to format your CSV in a special way. There's another option, which is to create an empty network and add your data manually. Here's a pseudo-script (you need to probably modify it since I didn't test it):

```python
import openpnm as op

net = op.network.GenericNetwork(coords=coords, conns=conns)
net["pore.diameter"] = Dp
net["throat.diameter"] = Dt
```

where `coords` is the xyz coordinates of your pores (Np by 3 array), `conns` is pore connectivity (i.e., which pore is connected to which -> Nt by 2 array), `Dp` is your pore diameter array, and `Dt` is throat diameter. Once you get this working, you need to add some more models to your network object to calculate other properties like throat length, etc. If you look into our examples, you'll see which models are required to estimate permeability.

**Reply by Soheil1993** [reply]
Dear Dr. @ma-sadeghi 

Thank you very much for your kind reply and valuable information.

I am a bit new to Python. Would you please give me more information in detail? Is there any similar example to follow?
Should I create all of these files separately before importing? In which format?
And how can I import them?

Many thanks in advance.

Best regards,
Soheil

**Reply by ma-sadeghi** [reply]
Well, `Dp` and `Dt` should be numpy arrays. You've mentioned that you have pore radii stored as CSV. You can use `pandas.read_csv` (see [here](https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html) for docs) to read your CSV file into a bunch of numpy arrays stored column-wise. You can google how to use `pandas.read_csv`, and you can find tons of examples.

**Reply by Soheil1993** [reply]
Dear Dr. @ma-sadeghi 

Thank you very much for your kind reply and useful link. 
I really appreciate your comprehensive description. 

Cheers, 
Soheil
